File: 0566-reshape-the-matrix/0566-reshape-the-matrix.js
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
    var arr = mat.flat();
  if (r * c != arr.length) return mat;

  var res = [];
  while (arr.length) res.push(arr.splice(0, c));
  return res;
};

File: 0566-reshape-the-matrix/0566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        flat = sum(mat, [])
        if len(flat) != r * c:
            return mat
        tuples = zip(*([iter(flat)] * c))
        return map(list, tuples)